Cancelling a task that is ended results in runtime panic

<!-- Please reserve GitHub issues for bug reports and feature requests.

For questions, the best place to get answers is on our [discussion forum](https://discuss.hashicorp.com/c/waypoint), as they will get more visibility from experienced users than the issue tracker.

Please note: We take Waypoint's security and our users' trust very seriously. If you believe you have found a security issue in Waypoint, please responsibly disclose by contacting us at security@hashicorp.com. Our PGP key is available at our security page: https://www.hashicorp.com/security/

-->

**Describe the bug**
Runtime panic on canceling a task that is (probably) ended.

**Steps to Reproduce**
1. Find a task in `waypoint-runner` assigned to the runner.
2. Try to cancel it with the following command.

Please include any `waypoint.hcl` files if applicable, as well as a
[GitHub Gist](https://gist.github.com/) of any relevant logs or steps to
reproduce the bug. Running `waypoint` commands with `-v` up to `-vvv` will
include any additional debugging info in the log.
```
(env) michal.weglowskihodur@mhodur1m-war1 psi % waypoint task cancel -run-job-id=01GEMEGQERGV02QQJVTWQT32SF -vv  
2022-10-10T11:10:18.939+0200 [INFO]  waypoint: waypoint version: full_string="v0.10.2 (2c9926ef8)" version=v0.10.2 prerelease="" metadata="" revision=2c9926ef8
2022-10-10T11:10:18.940+0200 [DEBUG] waypoint: home configuration directory: path=/Users/michal.weglowskihodur/Library/Preferences/waypoint
2022-10-10T11:10:18.946+0200 [INFO]  waypoint.server: attempting to source credentials and connect
2022-10-10T11:10:18.948+0200 [DEBUG] waypoint.serverclient: connection information: address=192.168.0.140:9701 tls=true tls_skip_verify=true send_auth=true has_token=true
2022-10-10T11:10:19.009+0200 [DEBUG] waypoint.server: connection established with sourced credentials
2022-10-10T11:10:19.019+0200 [INFO]  waypoint: server version info: version=v0.10.2 api_min=1 api_current=1 entrypoint_min=1 entrypoint_current=1
2022-10-10T11:10:19.019+0200 [INFO]  waypoint: negotiated api version: version=1

2022-10-10T11:10:19.019+0200 [WARN]  waypoint: context cancelled, stopping interrupt listener loop
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/hashicorp/waypoint/internal/cli.(*TaskCancelCommand).Run(0x14000dda3f0, {0x140002d07f0, 0x1, 0x1})
        /home/circleci/project/waypoint/internal/cli/task_cancel.go:33 +0x724
github.com/mitchellh/cli.(*CLI).Run(0x14000c71cc0)
        /home/circleci/go/pkg/mod/github.com/mitchellh/cli@v1.1.2/cli.go:262 +0x5c4
github.com/hashicorp/waypoint/internal/cli.Main({0x14000180050, 0x5, 0x5})
        /home/circleci/project/waypoint/internal/cli/main.go:127 +0x5cc
main.main()
        /home/circleci/project/waypoint/cmd/waypoint/main.go:14 +0xb4
```

**Expected behavior**
Message: `This job has been already finished`

**Waypoint Platform Versions**
Additional version and platform information to help triage the issue if
applicable:

* Waypoint CLI Version: v0.10.2 (2c9926ef8)
* Waypoint Server Platform and Version: (like `docker`, `nomad`, `kubernetes`): `nomad`
* Waypoint Plugin: (like `aws/ecs`, `pack`, `azure`): `?`

**Additional context**
I am now fighting with Waypoint on Nomad. The aim is to create self-deployable git repo with possibility of exposing to the internet. There was a problem with the installation of the waypoint runner in the home-lab. Needed to do some workarounds with  runner on nomad (the installer doesn't give all necessary config). But, after initial fights, it came to my attention that since waypoint server was running, there were loads and loads of tasks.

```
(env) michal.weglowskihodur@mhodur1m-war1 psi % waypoint task list
! grpc: received message larger than max (7870376 vs. 4194304)
(env) michal.weglowskihodur@mhodur1m-war1 psi % 
```
Ooooops.

--- a/.changelog/4019.txt
+++ b/.changelog/4019.txt
@@ -1,0 +1,4 @@
+```release-note:bug
+cli: Fix panic in cli for `waypoint task cancel` if attempting to cancel by run
+job id with no argument.
+```

--- a/internal/cli/task_cancel.go
+++ b/internal/cli/task_cancel.go
@@ -26,13 +26,15 @@ func (c *TaskCancelCommand) Run(args []string) int {
 	}
 	ctx := c.Ctx
 
-	if len(c.args) == 0 && c.flagRunJobId == "" {
+	var taskId string
+	if len(c.args) > 0 {
+		taskId = c.args[0]
+	}
+
+	if taskId == "" && c.flagRunJobId == "" {
 		c.ui.Output("Task ID or Run Job Id required.\n\n%s", c.Help(), terminal.WithErrorStyle())
 		return 1
-	}
-	taskId := c.args[0]
-
-	if c.flagRunJobId != "" && taskId != "" {
+	} else if c.flagRunJobId != "" && taskId != "" {
 		c.ui.Output("Both Run Job Id and Task Id was supplied, will look up by Task Id", terminal.WithWarningStyle())
 	}
 

--- a/internal/cli/task_cancel.go
+++ b/internal/cli/task_cancel.go
@@ -32,7 +32,7 @@ func (c *TaskCancelCommand) Run(args []string) int {
 	}
 
 	if taskId == "" && c.flagRunJobId == "" {
-		c.ui.Output("Task ID or Run Job Id required.\n\n%s", c.Help(), terminal.WithErrorStyle())
+		c.ui.Output("Task Id or Run Job Id required.\n\n%s", c.Help(), terminal.WithErrorStyle())
 		return 1
 	} else if c.flagRunJobId != "" && taskId != "" {
 		c.ui.Output("Both Run Job Id and Task Id was supplied, will look up by Task Id", terminal.WithWarningStyle())

